src/Test/Utilities/Portable/TempFiles/TempRoot.cs

```diff
@@ -20,16 +20,7 @@ static TempRoot()
 
         public void Dispose()
         {
-            if (_temps != null)
-            {
-                DisposeAll(_temps);
-                _temps.Clear();
-            }
-        }
-
-        private static void DisposeAll(IEnumerable<IDisposable> temps)
-        {
-            foreach (var temp in temps)
+            while (_temps.TryTake(out var temp))
```

Comment:
In general Dispose is a well known state transition for an object from usable to dead. This code though is allowing mutations, after dispose, to silently work.